frontend/public/components/secrets/create-secret.tsx

```diff
@@ -419,22 +428,53 @@ class CreateConfigSubform extends React.Component<CreateConfigSubformProps, Crea
   checkEntryAddressForDuplicates(updatedData, updatedIndex) {
     return this.state.secretEntriesArray.some((entry, i) => i !== updatedIndex && _.isEqual(entry.address, updatedData.address));
   }
-  onDataChanged (updatedEntryData, entryID) {
+  propagateEntryChange(secretEntriesArray) {
+    const imageSecretObject = this.imageSecretArrayToObject(secretEntriesArray);
+    this.props.onChange(this.state.isDockerconfigjson ? {[AUTHS_KEY]: imageSecretObject} : imageSecretObject);
+  }
+  onDataChanged(updatedEntryData, entryID) {
     const secretEntriesArray = this.state.secretEntriesArray;
     secretEntriesArray.splice(entryID, 1, updatedEntryData);
     this.setState({
       secretEntriesArray: secretEntriesArray,
       hasDuplicate: this.checkEntryAddressForDuplicates(updatedEntryData, entryID)
     }, () => {
-      const imageSecretObject = this.imageSecretArrayToObject(this.state.secretEntriesArray);
-      this.props.onChange(this.state.isDockerconfigjson ? {[AUTHS_KEY]: imageSecretObject} : imageSecretObject);
+      this.propagateEntryChange(this.state.secretEntriesArray);
+    });
+  }
+  removeEntry(entryID, e){
+    e.preventDefault();
+    const secretEntriesArray = this.state.secretEntriesArray;
+    secretEntriesArray.splice(entryID, 1);
+    this.setState({
+      secretEntriesArray: secretEntriesArray
+    }, () => {
+      this.propagateEntryChange(this.state.secretEntriesArray);
+    });
+  }
+  addEntry(e){
+    e.preventDefault();
+    this.setState({
+      secretEntriesArray: _.concat(this.state.secretEntriesArray, this.newImageSecretEntry())
+    }, () => {
+      this.propagateEntryChange(this.state.secretEntriesArray);
     });
   }
   render() {
-    const secretEntriesList = _.map(this.state.secretEntriesArray, (entry, index) => <ConfigEntryForm key={index} id={index} entry={entry} onChange={this.onDataChanged} />);
+    const secretEntriesList = _.map(this.state.secretEntriesArray, (entry, index) => {
+      return <div className="co-create-image-secret__form-wrapper" key={`${entry.uid}`}>
```

Comment:
Read that doc, but thought that since there are two siblings, would be better to hook them up with proper key names - `${entry.uid}-form` & `${entry.uid}-link`, since they will will have to be different anyways
But they can be names as suggested :)